cli/democluster: TestTransientClusterMultitenant failed

cli/democluster.TestTransientClusterMultitenant [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4539533&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4539533&tab=artifacts#/) on master @ [5755663effbdbe8175bb74e5cdc19b2298e0a0bc](https://github.com/cockroachdb/cockroach/commits/5755663effbdbe8175bb74e5cdc19b2298e0a0bc):


```
=== RUN   TestTransientClusterMultitenant
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/b296432f77c11d14af9987c5728df5a9/logTestTransientClusterMultitenant3290529205
    test_log_scope.go:80: use -show-logs to present logs inline
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterMultitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13670

```
panic: runtime error: integer divide by zero

goroutine 18107 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).partitionSpansTenant(0xc009a2e140, 0xc004a4a2d0, {0xc0012a0c60, 0x3, 0x3})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1165 +0x63d
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PartitionSpans(0x2, 0x1, {0xc0012a0c60, 0x2, 0xc00432cb68})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:991 +0x3c
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).maybeParallelizeLocalScans(0xc009a2e140, 0xc004a4a2d0, 0xc00432d058)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1417 +0x10e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planTableReaders(0xc009a2e140, 0x0, 0xc00869ba80, 0xc00432d058)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1499 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createTableReaders(0x1, 0xc004a4a2d0, 0xc0048d8620)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1342 +0x2db
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPhysPlanForPlanNode(0xc009a2e140, 0xc004a4a2d0, {0x533e628, 0xc0048d8620})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:3046 +0x631
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPhysPlanForPlanNode(0xc009a2e140, 0xc004a4a2d0, {0x533e468, 0xc008542000})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:3036 +0x4cc
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPhysPlan(0xc00432e6c0, 0x1, {{0x533e468, 0xc008542000}, 0x0})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2964 +0x58
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0x533c718, {0x533c718, 0xc00a9fd020}, 0x1, 0xc004a4a2d0, 0xc00a9fce70, {{0x533e468, 0xc008542000}, 0x0}, 0xc00d875500)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1421 +0x118
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00a99e000, {0x533c718, 0xc00a9fd020}, 0xc00a99e6b8, 0xc00a9fd020, {0x53c6648, 0xc00a9fce70}, 0x0, 0xc00e008838)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1442 +0x605
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00a99e000, {0x533c718, 0xc00a9fd020}, 0xc00a99e6b8, {0x53c6648, 0xc00a9fce70})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1118 +0x9e5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00a99e000, {0x533c670, 0xc003ecc400}, {{0x536a458, 0xc0013e0780}, {0x4286730, 0x7d}, 0x1, 0x1}, 0xc004271a00, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:685 +0x20a5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00a99e000, {0x533c670, 0xc003ecc400}, {{0x536a458, 0xc0013e0780}, {0x4286730, 0x7d}, 0x1, 0x1}, 0xc004271a00, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:143 +0x57e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc00a99e000, {0x533c670, 0xc003ecc400}, {0xc004271a00, {0xc00d819d70, 0x1, 0x1}, {0x0, 0x0, 0x0}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:231 +0x16e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2({{0x0, 0x0}, 0x0, {0xbaa0514, 0xed9bba255, 0x0}, 0x0}, 0xc00a99e000, 0xc00432fb80, 0xc00432fb70, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1924 +0x552
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00a99e000)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1926 +0x41b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00a99e000, {0x533c718, 0xc00a9fc210}, 0xc00a6da4e0, {0xc0004ab7d0, 0xa7c678, 0x0, 0x0}, 0xc001664d00)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1779 +0x245
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx.func1()
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:205 +0x97
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:204 +0x5f1
```

```
panic: runtime error: integer divide by zero

goroutine 18536 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).partitionSpansTenant(0xc006937e00, 0xc0032991d0, {0xc008f4c750, 0x3, 0x3})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1165 +0x63d
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PartitionSpans(0x7, 0x5, {0xc008f4c750, 0x0, 0xc007cca9a0})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:991 +0x3c
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).maybeParallelizeLocalScans(0xc006937e00, 0xc0032991d0, 0xc00d4e7050)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1417 +0x10e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planTableReaders(0xc006937e00, 0x0, 0xc00d4b0580, 0xc00d4e7050)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1499 +0x52
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createTableReaders(0x54566f0, 0xc0032991d0, 0xc0099f8a80)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:1342 +0x2db
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPhysPlanForPlanNode(0xc006937e00, 0xc0032991d0, {0x53a88a0, 0xc0099f8a80})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2987 +0x631
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).createPhysPlanForPlanNode(0xc006937e00, 0xc0032991d0, {0x53a86e0, 0xc00d492b40})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go:2977 +0x4cc
...
```

--- a/pkg/sql/distsql_physical_planner.go
+++ b/pkg/sql/distsql_physical_planner.go
@@ -1149,6 +1149,9 @@ func (dsp *DistSQLPlanner) partitionSpansTenant(
 	if err != nil {
 		return nil, err
 	}
+	if len(instances) == 0 {
+		return nil, errors.New("no healthy sql instances available for planning")
+	}
 	// Randomize the order in which we assign partitions, so that work is
 	// allocated fairly across queries.
 	rand.Shuffle(len(instances), func(i, j int) {

